### ISSUE TITLE
MAME ROMs won't load after the first time

### ISSUE THREAD

**Issue by Raziel480** [OpenEmu/OpenEmu#3149]
tl;dr If using experimental Arcade, do NOT allow automatic loading of the previous save state. ROMs will not load after the first time.

Details:
Using the current experimental build, I discovered an issue attempting to start a ROM any time after the first time in OpenEmu. The first time I run a ROM it is fine, but every time after the first time I just get a black screen. When this happens I have to force quit OpenEmu. ROMs will not open a second time.

I tracked down the cause. When emulation is stopped it automatically creates a save state. Something about MAME or OpenEmu does not like how save states are loaded at startup. If you do not load a save automatically at startup the ROM will open. From there you can manually load the save. The issue is not with the automatically generated save file. I used the Secrets menu to show manual and automatic saves. I can load either without issue after emulation has started.

If you already told OpenEmu to load states without asking, you can use the preferences to resetting warnings. It will then ask again before the next time you try and load the ROM. From there say No when it asks to load a save at startup. Once emulation starts you can manually load the save state.

Steps to reproduce
1. Add new ROM to OpenEmu (I used Tapper (rev 2 I think) tappera.zip)
2. Launch ROM
3. Let run a few moments
4. Stop emulation (or quit OpenEmu)
5.  Re-launch ROM

Expected results:
1. ROM loads

Actual results:
1. Black window and ROM does not load.
2. OpenEmu "soft" freezes. Windows can be moved and you can click buttons but nothing happens.
3. Quitting OpenEmu does not work. I had to Force Quit

**Comment by clobber** [comment]
https://github.com/OpenEmu/OpenEmu/issues/2615#issuecomment-230357101

**Comment by clobber** [comment]
Okay, I see you were working on a workaround. If you have something that makes sense, open a Pull Request for review.

**Comment by Raziel480** [comment]
Yeah I know it's unsupported. I wrote it up so others might find the workaround. I'm working on a fix for it for myself. Once I have one should I try and make a pull request or is it not accepting submissions?

**Comment by Raziel480** [comment]
I guess we were writing at the same time. I saw your note. I'll play with it a bit more and submit a pull request once it's ready.

**Comment by clobber** [comment]
One thing to note, restoring save states at launch *used* to work with the MAME core plugin, so it might be worth tracking down where that regression started. I can't remember the version off the top of my head, but you can play around with older releases if you want: https://github.com/OpenEmu/OpenEmu/releases

**Comment by clobber** [comment]
Okay, this may have been introduced in 2.0.2. Check the refactorings that occurred in commits during March 2016, could possibly be related to the XPC changes -- hope that's useful.

**Comment by Raziel480** [comment]
I'll try and take a look tomorrow. I confirmed it worked in 2.0.1 and stopped working in 2.0.2. Oddly after I open 2.0.2 I can't open 2.0.1 anymore. 2.0.1 crashes with "“%@” can’t be found." I see "Library not loaded: @rpath/libswiftAppKit.dylib" in the crash report for it. Restarting seems to get me around that. I'll see if I can find the regression. If not I'll go with the original fix I was working on.

**Comment by clobber** [comment]
Moved to https://github.com/OpenEmu/UME-Core/issues/7

**Comment by BenitolSantos** [comment]
Is this considered a workaround? https://github.com/OpenEmu/OpenEmu/issues/2589
